src/controllers/shared.controllers/director.correspondence.address.link.controller.ts

```diff
@@ -4,29 +4,30 @@ import { createValidationErrorBasic, formatValidationErrors } from '../../valida
 import { OfficerFiling } from "@companieshouse/api-sdk-node/dist/services/officer-filing";
 import { getOfficerFiling, patchOfficerFiling } from "../../services/officer.filing.service";
 import { formatTitleCase } from "../../services/confirm.company.service";
-import { retrieveDirectorNameFromFiling } from "../../utils/format";
+import { retrieveDirectorNameFromAppointment, retrieveDirectorNameFromFiling } from "../../utils/format";
 import { DirectorField } from "../../model/director.model";
-import { getField } from "../../utils/web";
+import { getDirectorNameBasedOnJourney, getField } from "../../utils/web";
 import { Session } from "@companieshouse/node-session-handler";
 import { selectLang, getLocalesService, getLocaleInfo } from "../../utils/localise";
 import { saToRoaErrorMessageKey } from "../../utils/api.enumerations.keys";
 import { CompanyAppointment } from "private-api-sdk-node/dist/services/company-appointments/types";
 import { compareAddress } from "../../utils/address";
 import { getCompanyAppointmentFullRecord } from "../../services/company.appointments.service";
 
-export const getCorrespondenceLink = async (req: Request, res: Response, next: NextFunction, templateName: string, backUrlPath: string) => {
+export const getCorrespondenceLink = async (req: Request, res: Response, next: NextFunction, templateName: string, backUrlPath: string, isUpdate: boolean) => {
   try {
     const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
     const submissionId = urlUtils.getSubmissionIdFromRequestParams(req);
     const session: Session = req.session as Session;
     const locales = getLocalesService();
     const lang = selectLang(req.query.lang);
     const officerFiling = await getOfficerFiling(session, transactionId, submissionId);
+    const directorName = await getDirectorNameBasedOnJourney(isUpdate, session, req, officerFiling);
```

Comment:
The same change missing in the postCorrespondenceLink around line 59 when rendering a validation error.